jcl/src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/GenerateJLIClassesPlugin.java

```diff
@@ -19,4 +19,49 @@
  * [2] http://openjdk.java.net/legal/assembly-exception.html
  *
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
- *******************************************************************************/
\ No newline at end of file
+ *******************************************************************************/
+package jdk.tools.jlink.internal.plugins;
+
+import jdk.tools.jlink.plugin.Plugin;
+import jdk.tools.jlink.plugin.ResourcePool;
+import jdk.tools.jlink.plugin.ResourcePoolBuilder;
+import jdk.tools.jlink.plugin.ResourcePoolEntry;
+import java.util.function.Function;
+
+/**
+ * We override the default GenerateJLIClassesPlugin with this
+ * one because the default relies on Hotspot-specific code.
+ * Removing this class entirely causes the build to fail when
+ * using the default module-info.java file for the jlink module.
+ * So we create the class, but include no code that affects any 
+ * change.
+ */
+public class GenerateJLIClassesPlugin implements Plugin
+{
+ /**
+   * This method makes a copy of the ResourcePool.
+   * Since this plugin is intended to have no effect,
+   * it lacks any transformational logic.
+   * @param rp            Pool of resources
+   * @param rpb           Builder to create a copy of the pool of resources.
+   * @return ResourcePool Built copy of the pool of resources.
+   */
+   public ResourcePool transform(ResourcePool rp, ResourcePoolBuilder rpb){
+      Function<ResourcePoolEntry, ResourcePoolEntry> returnsArg = (val) -> {
```

Comment:
Considering the plugin is disabled, can the implementation here be `return null;`?